pywr/_core.pyx

```diff
@@ -800,19 +858,54 @@ cdef class StorageInput(BaseInput):
         BaseInput.commit(self, scenario_index, volume)
         self._parent.commit(scenario_index, -volume)
 
+    cpdef commit_all(self, double[:] value):
```

Comment:
I'm guessing these method are just to allow net flow for a storage node to be easily recorded?

Reply:
It allows the solver to set net flows for all scenarios simultaneously. A `commit_all` method already exists on `AbstractNode`, but these storage nodes need different behaviour to update the parent storage node.